### PR TITLE
Add layout_calendar

### PR DESCRIPTION
**Pull request by YegorKochetkov**
- [DEMO LINK](https://YegorKochetkov.github.io/layout_calendar/)
- [TEST REPORT LINK](https://YegorKochetkov.github.io/layout_calendar/report/html_report/)

### CODE REVIEW

File: src/styles/utils/variables.scss [hunk 1]
```diff
@@ -0,0 +1,10 @@
+$dayOfWeek:
+  "mon" 1, "tue" 2, "wen" 3, "thu" 4, "fri" 5, "sat" 6, "sun" 7;
```
Comment (ivan-kadykalo) [inline src/styles/utils/variables.scss]: I heard that the each value should be in a new line, but i'm not sure is it necessarily or not.

**Review by ivan-kadykalo — APPROVED** [review]
Looks like everithing is ok. Good job!

File: src/styles/utils/variables.scss (hunk 1 above)
Comment (YegorKochetkov) [reply src/styles/utils/variables.scss]: very likely

File: src/index.html [hunk 2]
```diff
@@ -1,12 +1,44 @@
...
     <meta charset="UTF-8">
     <meta name="viewport" content="width=device-width, initial-scale=1.0">
     <title>Calendar</title>
     <link rel="stylesheet" href="./styles/main.scss">
   </head>
-  <body>
-    <h1>Calendar</h1>
+  <body class="page__content">
+    <article class="calendar calendar--start-day--sun calendar--month-length--29">
+      <p class="day day--1"></p>
```
Comment (illiarerun) [inline src/index.html]: i'm pretty sure that `day` doesn't have to be a block, you won't reuse it outside of calendar anyway, so make it an element of calendar block

Comment (illiarerun) [reply src/index.html]: also, use nth-child, don't use modificators here, imagine if you needed to create 100 days calendar? it is easy to make a mistake with modificator names

File: src/styles/utils/variables.scss [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+  "sat" 6,
+  "sun" 7;
+
+$dayCellSize: 100px;
+$dayCellInRow: 7;
+$dayGap: $dayCellSize / 100;
+$dayCellTranslateY: $dayCellSize / 5;
+
+$calendarPaddings: $dayCellSize / 10;
+$calenarWidth:($dayCellSize + $dayGap) * $dayCellInRow + $calendarPaddings * 2;
```
Comment (illiarerun) [inline src/styles/utils/variables.scss]:
```suggestion
$calendarWidth:($dayCellSize + $dayGap) * $dayCellInRow + $calendarPaddings * 2;
```

Comment (illiarerun) [reply src/styles/utils/variables.scss]: there are only 6 gaps between 7 day blocks

File: src/styles/reset.scss [hunk 4]
```diff
@@ -0,0 +1,3 @@
+* {
+  box-sizing: border-box;
+}
```
Comment (illiarerun) [inline src/styles/reset.scss]: it's a normalize style

File: src/styles/blocks/day.scss [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+  justify-content: center;
+  align-items: center;
+
+  @include square($dayCellSize);
+  margin: 0;
+
+  background-color: #eee;
+  border: 1px solid #000;
+
+  transition-duration: 0.5s;
```
Comment (illiarerun) [inline src/styles/blocks/day.scss]: you should specify the properties that will change
```suggestion
  transition-duration: background-color 0.5s, transform 0.5s;
```

File: src/styles/blocks/calendar.scss [hunk 6]
```diff
@@ -0,0 +1,22 @@
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  gap: $dayGap;
+
+  max-width: $calenarWidth;
+  margin: auto;
+  padding: $calendarPaddings;
+  padding-top: $dayCellSize;
```
Comment (illiarerun) [inline src/styles/blocks/calendar.scss]: padding: $dayCellSize $calendarPaddings 0;

**Review by illiarerun — CHANGES_REQUESTED** [review]
Good job 🔥 Now let's work on fixes:

**Review by illiarerun — APPROVED** [review]

**Pull request closed without merging**